gloo/types.h

```diff
@@ -56,17 +56,27 @@ struct __attribute__((__aligned__(2))) float16 {
     return x == res.x;
   }
 #ifdef __CUDA_ARCH__
-  float16(half h) {
+  float16(__half h) {
```

Comment:
I'm not 100% sure either :) I've always known the type as `__half`, and that's what is defined in cuda_fp16.h